### ISSUE TITLE
Mod Doesn't Properly Consider Armor Penetration Revamp

### ISSUE THREAD

**Issue by maarxx** [notfood/RimWorld-Outfitted#10]
# Mod Doesn't Properly Consider Armor Penetration Revamp

## Short Version

Since the revamp, it doesn't matter so much what the additive sum of `Armor-Sharp` is across all pieces, In general when maximizing `Armor-Sharp`, the most important thing is to have the highest possible value on a single piece / in a single layer, and then simply fill in the rest as best you can.

## Long Version

I don't think this mod properly considers armor since the Armor Penetration revamp of B19. I'm talking primarily about `Armor-Sharp` stat, since this is what is used for bullets, and is commonly the most important (sometimes only) stat considered for combat survivability.

I noticed my pawns often prefer to wear a set of Flak (Flak Vest + Flak Jacket + Flak Pants) over, say, Marine Armor.

I think this is because the pieces, when simply summed up, offer more points of `Armor-Sharp` than the Marine Armor, and this is how armor _used_ to work (B18 and prior), but this isn't how Armor actually works in combat anymore.

The system was revamped in B19 for this reason.

https://ludeon.com/blog/2018/08/rimworld-beta-19-released/
> Armor system reworked into a 3-outcome system with armor penetration. The armor penetration of the weapon is subtracted from the armor value of the armor. This gives an effective armor rating, measured as a percent. We roll a RNG out of 100%. If the result is under half the effective AR, the hit is totally deflected. If the result is over half the effective AR, but under the effective AR, the hit is mitigated – damage is reduced by half and sharp hits are converted to bruise. If the result is over the effective AR, the hit passes the armor and applies normally. As before, armor is applied in sequence from the outside in. AR maxes out at 200%. Default armor penetration is 1.5% per point of damage, but some weapons vary from this.

https://rimworldwiki.com/wiki/Armor#Armor_rating
> Each effective point of armor gives [...]

> Armor is calculated per piece of apparel, from the outside in. Armor mitigation does stack multiplicatively, e.g. 2 pieces of clothing or armor will mitigate damage twice for 25% damage. Armor Penetration is also applied on each layer, this means that apparel with low armor values will do little or nothing against projectiles with high armor penetration.

> For example a pawn with both a devilstrand shirt with sharp armor of 28% and a toughskin gland with sharp armor 35% would always receive full damage from a charge rifle, as its armor penetration of 35% completely nullifies the protection of both items. This is why higher armor values are generally preferred over multiple layers of much lower armor values. However there is little reason not to stack the best protection on your pawns when the items not mutually exclusive. A hyperweave shirt might still stop an assault rifle bullet that penetrates a flak vest, and it could be the only thing protecting the pawns arm.

So in general, since the revamp, it doesn't matter so much what the additive sum of `Armor-Sharp` is across all pieces, In general when maximizing `Armor-Sharp`, the most important thing is to have the highest possible value on a single piece / in a single layer, and then simply fill in the rest as best you can.

**Comment by Zelacks** [comment]
Agreed. I had this same problem. Looking at the mod though, it doesn't really seem to do anything super special for each stats, it treats them all identically, which is why its a problem since the armour has special requirements.

**Not a solution,** but I had success getting pawns equipping what I want by increase the weighting of market value. The good pants and shirts (hyperweave, devilstrand etc) are all high value anyway so they will be equipped, but also it makes marine/recon etc armour much more desirable as they are much more expensive than anything else. 
In my example, I had two outfits, peace time outfit that forbids wearing the armours, and a wartime outfit that has all the armours. When war starts, I set all the pawns I want to equip the armours to the wartime outfit, and using the market value weighting, they all equip the more desirable armours first.

**Comment by maarxx** [comment]
The problem is in the transition from Flak to power armors.

If the outfit only allows Flak, they pick the right pieces of Flak.

If the outfit only allows one-piece power armors, they pick the best kinds.

The problem only exists in the transition.

Therefore, as my workaround, I would just create two different outfits, one for each type (one that allows Flak and one that disallows Flak), and before assigning out the policies, manually count (Alert Everything) how many power armors I had, and dole out the right number of non-flak policies, and give everyone else the Flak policy.

This turns out to not be a huge deal anyway, because if power armor is limited, I've usually got preferred pawns I would like to wear it.

Once you have enough power armor for everyone, just delete your Flak outfit policy.
